store/sqlstore/post_store.go

```diff
@@ -1780,3 +1790,34 @@ func (s *SqlPostStore) SearchPostsInTeamForUser(paramsList []*model.SearchParams
 
 	return model.MakePostSearchResults(posts, nil), nil
 }
+
+func wrapEmails(terms string) (string, bool) {
+	emailRegex := regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
```

Comment:
No need to have a perfect regexp here, with the risk that implies, we are using a simpler one [here](https://github.com/mattermost/mattermost-server/blob/master/services/searchengine/utils.go#L13). You can use something similar